core/iwasm/libraries/wasi-nn/wasi_nn.c

```diff
@@ -5,64 +5,61 @@
 #include <string.h>
 #include <stdlib.h>
 
-#include <tensorflow/lite/interpreter.h>
-#include <tensorflow/lite/kernels/register.h>
-#include <tensorflow/lite/model.h>
-#include <tensorflow/lite/optional_debug_tools.h> 
+#include "wasm_export.h"
 
-std::unique_ptr<tflite::Interpreter> interpreter = NULL;
-std::unique_ptr<tflite::FlatBufferModel> model =NULL;
+// #include <tensorflow/lite/interpreter.h>
+// #include <tensorflow/lite/kernels/register.h>
+// #include <tensorflow/lite/model.h>
+// #include <tensorflow/lite/optional_debug_tools.h> 
 
-º
-void load(graph_builder_array builder, graph_encoding encoding)
-{
-    model = tflite::FlatBufferModel::BuildFromBuffer(builder[0]);
-    tflite::ops::builtin::BuiltinOpResolver resolver;
-	tflite::InterpreterBuilder builder(*model, resolver);
-	builder(&interpreter);
+// std::unique_ptr<tflite::Interpreter> interpreter = NULL;
+// std::unique_ptr<tflite::FlatBufferModel> model = NULL;
 
+void wasi_nn_load(wasm_exec_env_t exec_env, graph_builder_array builder, graph_encoding encoding)
+{
+    // tflite::ErrorReporter *error_reporter;
+    // model = tflite::FlatBufferModel::BuildFromBuffer(
+    //     (const char *)builder[0],
+    //     1000, // TODO: find how to pass buffer size
+    //     error_reporter
+    // );
+    // tflite::ops::builtin::BuiltinOpResolver resolver;
+	// tflite::InterpreterBuilder(*model, resolver)(&interpreter);
+    printf("Inside wasi_nn_load!\n\n");
```

Comment:
I commented everything since C++ is incompatible directly to WAMR. The function calls to C++ will be placed in a separate file with `extern "C"`.